src/js/lib/youtube.js

```diff
@@ -4,34 +4,45 @@ import reqwest from 'reqwest';
 export function pimpYouTubePlayer(videoId, node, height, width) {
     youTubeIframe.init(function() {
         //preload youtube iframe API
-        node.querySelector('#ytGuPlayer').addEventListener('click', function() {
-            var youTubePlayer = youTubeIframe.createPlayer(this, {
+        var promise = new Promise(function(resolve) {
+            var youTubePlayer = youTubeIframe.createPlayer(node.querySelector('#ytGuPlayer'), {
```

Comment:
I'd make this a `const` rather than a `var`, especially as its being passed around now.